packages/theia-integration/src/node/glsp-socket-server-contribution.ts

```diff
@@ -41,6 +43,9 @@ export interface GLSPSocketServerContributionOptions extends GLSPServerContribut
 
     /** Additional arguments that should be passed when starting the server process. */
     additionalArgs?: string[];
+
+    /** Path of the WebSocket GLSP Server. */
+    webSocketPath?: string;
```

Comment:
I think it would be a nice if we would introduce an abstract `GlspJsonRpcServerLauncher` super class here and then introduce dediacted subclasses for the socket & websocket use case. The orthogonal [PR ](https://github.com/eclipse-glsp/glsp-theia-integration/pull/154) for running the GLSP server directly in the node backend could also use this super class.

Reply:
That sounds good in general! As discussed with you offline, I will create a follow-up ticket for that.
Because in my use case, I can consume everything from the existing `GLSPSocketServerContribution` and only have a different behaviour in the `connect` function in case of the WebSocket communication.